cfn-resources/x509-authentication-database-user/cmd/resource/resource.go

```diff
@@ -18,89 +18,75 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"log"
 
 	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/cloudformation"
-	"github.com/mongodb/mongodbatlas-cloudformation-resources/profile"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
-	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
 	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas-sdk/v20230201008/admin"
 	"go.mongodb.org/atlas/mongodbatlas"
 )
 
 var CreateRequiredFields = []string{constants.ProjectID, constants.UserID}
 var ReadRequiredFields = []string{constants.ProjectID}
 
-// Create handles the Create event from the Cloudformation service.
-func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
-	setup() // logger setup
-	log.Print(currentModel.ProjectId)
-	log.Print(currentModel.UserName)
-	modelValidation := validateModel(CreateRequiredFields, currentModel)
-	if modelValidation != nil {
-		return *modelValidation, nil
-	}
+func setup() {
+	util.SetupLogger("mongodb-atlas-x509-authentication-database-user")
+}
 
-	if currentModel.Profile == nil {
-		currentModel.Profile = aws.String(profile.DefaultProfile)
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+	util.SetDefaultProfileIfNotDefined(&currentModel.Profile)
+	if err := validator.ValidateModel(CreateRequiredFields, currentModel); err != nil {
+		return *err, nil
 	}
 
-	// Create atlas client
-	client, peErr := util.NewMongoDBClient(req, currentModel.Profile)
-	if peErr != nil {
-		return *peErr, nil
+	client, pe := util.NewAtlasClient(&req, currentModel.Profile)
+	if pe != nil {
+		return *pe, nil
 	}
 
-	// progress callback setup
-	if _, ok := req.CallbackContext["status"]; ok {
-		sid := req.CallbackContext["ProjectId"].(string)
-		currentModel.ProjectId = &sid
-		return validateProgress(client, currentModel)
```

Comment:
just to be sure, this was completely removed as there is not CallbackContext defined in returned progress events right?

Reply:
exactly, the context is never set to continue